### PR TITLE
Add server side rendering

### PR DESCRIPTION
**Pull request by arielsvn**
## Issue Number

#844 

## Purpose/Implementation Notes

This turned out to be more complicated than I was expecting, however, I'm happy with how it's coming along. I deployed a demo to https://refinebio.now.sh/

A few things are missing from this PR in order to be ready:

- [x] Find a solution to navigate between pages that don't rely on the URL `state` because NextJS doesn't support this by default.

- [x] Fix filters on search

- [x] Fix unit tests

- [x] Fix integration tests

- [x] Compendia page toggle

- [x] Remove all packages related to CRA

- [ ] Add deploy configuration, we talked about Lambda@Edge. Alternatively, now.sh also integrates with Github and can deploy the app automatically.

@davidsmejia feel free to take a look, I'd really appreciate any comments. With the number of changes this includes we'll have to review it carefully. 

## Types of changes

* [ ] Bugfix (non-breaking change which fixes an issue)
* [ ] New feature (non-breaking change which adds functionality)
* [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

## Functional tests

Demo on https://refinebio.now.sh/

## Checklist

* [ ] Lint and unit tests pass locally with my changes
* [ ] I have added tests that prove my fix is effective or that my feature works
* [ ] I have added necessary documentation (if appropriate)
* [ ] Any dependent changes have been merged and published in downstream modules

### CODE REVIEW

**Comment by arielsvn** [comment]
@davidsmejia I checked this a couple more times, it should be ready for review. I'm going to update the readme tomorrow to reflect the nextjs changes and also want to investigate if we can improve the code splitting, right now looks like it's generating too many files.

File: .gitignore [hunk 1]
```diff
@@ -21,5 +24,5 @@ npm-debug.log*
 yarn-debug.log*
 yarn-error.log*
 
-# vim
```
Comment (davidsmejia) [inline .gitignore]: oops :/

**Comment by davidsmejia** [comment]
Also we should figure out previews/deploy before merging. Or at least have a roadmap in place

File: .gitignore (hunk 1 above)
Comment (arielsvn) [reply .gitignore]: haha not sure why that got commented out :)

Comment (davidsmejia) [reply .gitignore]: oh it's just a header explaining the following lines... so i don't accidentally commit my swap files that are generated by vim

Comment (davidsmejia) [reply .gitignore]: i did not mean to add the trailing white-space though

**Comment by arielsvn** [comment]
Talked with @davidsmejia and we came up with a plan to deploy this:

1. Address the comments in this PR and get it merged
2. Create a CCDL team in Zeit, and setup both staging and prod deploys under `.now.sh` domains
3. On Tuesday, when I'm back, point www.refine.bio and staging.refine.bio to the ZEIT servers.
4. Remove all frontend related terraform configuration from refinebio's backend.

File: src/pages/search/ResultFilters/FilterCategory.js [hunk 2]
```diff
@@ -68,7 +68,7 @@ export function FilterCategory({
             setSavedQuery(query);
             onToggleFilter(queryField, filter);
           }}
-          checked={activeValues && activeValues.includes(filter)}
+          checked={!!activeValues && activeValues.includes(filter)}
```
[the remark this answers is not in the document]
Comment (arielsvn) [reply src/pages/search/ResultFilters/FilterCategory.js]: ah `!!` has the same effect and it's used extensively in the project. I didn't know you could also do `Boolean(`

**Pull request merged**